### PR TITLE
snapshotting: remove files where possible

### PR DESCRIPTION
**Pull request by danp**
If errors happen during snapshotting a lot of temp/etc files can be left around. Delete them when possible.

Also clean up a couple minor things.

### CODE REVIEW

**Comment by danp** [comment]
Might be easier to view diff [here](https://github.com/sclasen/streambolt/pull/10/files?w=1).

File: snapshotter.go [hunk 1]
```diff
@@ -85,41 +84,38 @@ func (s *ShardSnapshotter) SnapshotShard() (*Snapshot, error) {
 		latest = l
 	}
 
-	if latest != nil {
```
Comment (danp) [inline snapshotter.go]: There didn't seem to be a way for latest to be nil at this point if above bits didn't error.

File: snapshotter.go [hunk 2]
```diff
@@ -239,8 +235,8 @@ func (s *ShardSnapshotFinder) DownloadSnapshot(snapshot Snapshot) error {
 }
 
 func (s *ShardSnapshotter) ToWorkingCopy(snapshot Snapshot) (string, error) {
-	copy := (fmt.Sprintf("%s/working-%s-%s-%d", s.LocalPath, s.Stream, s.ShardId, time.Now().UnixNano()))
-	return copy, exec.Command("mv", snapshot.LocalFile, copy).Run()
+	copy := fmt.Sprintf("%s/working-%s-%s-%d", s.LocalPath, s.Stream, s.ShardId, time.Now().UnixNano())
+	return copy, os.Rename(snapshot.LocalFile, copy)
```
Comment (danp) [inline snapshotter.go]: Changed to os.Rename here and below, don't think there was any special reason to shell out to mv.

**Comment by sclasen** [comment]
👍

**Pull request merged**